src/redux/selectors/index.js

```diff
@@ -196,16 +178,140 @@ export const patientAgeAtResourcesSelector = createSelector(
+);
+
+export const collectionResourceIdsSelector = createSelector(
+  [collectionsSelector, selectedCollectionSelector],
+  (collections, selectedCollectionId) => collections[selectedCollectionId].resourceIds,
+);
+
+const selectedCollectionResourceIdsSelector = createSelector(
+  [collectionsSelector, selectedCollectionSelector],
+  (collections, selectedCollection) => collections[selectedCollection]?.resourceIds,
+);
 
-      acc[id] = ageAtResourceDate;
+const isResourceInDateRange = (resource, dateRangeStart, dateRangeEnd) => {
+  const { timelineDate } = resource;
+  if (!timelineDate) {
+    console.info('record does not have an timelineDate, resourceId: ', resource.id); // eslint-disable-line no-console
+    return false;
+  }
+  if (!dateRangeStart || !dateRangeEnd) { return true; }
+  return (
+    isWithinInterval(
+      timelineDate, { start: startOfDay(dateRangeStart), end: endOfDay(dateRangeEnd) },
+    )
+  );
+};
+
+const filteredResourceTypesSelector = createSelector(
+  [
+    resourceTypeFiltersSelector,
+    resourceIdsGroupedByTypeSelector,
+    selectedResourceTypeSelector,
+    selectedCollectionResourceIdsSelector,
+    dateRangeFilterFiltersSelector,
+    resourcesSelector,
+    timelinePropsSelector,
+  ],
+  (
+    resourceTypeFilter,
+    resourceIdsGroupedByType,
+    selectedResourceType,
+    selectedCollectionResourceIdsObjects,
+    dateRangeFilterFilters,
+    resources,
+    timelineProps,
+  ) => {
+    if (!selectedCollectionResourceIdsObjects) {
+      return {};
+    }
+    const { minimumDate, maximumDate } = timelineProps;
+    const { dateRangeStart, dateRangeEnd } = dateRangeFilterFilters;
+    const activeDateStart = dateRangeStart || minimumDate;
+    const activeDateEnd = dateRangeEnd || maximumDate;
+    const selectedCollectionResourceIds = Object.keys(selectedCollectionResourceIdsObjects);
+    return Object.keys(resourceTypeFilter).reduce((acc, resourceType) => {
+      if (resourceTypeFilter[resourceType]) {
+        acc[resourceType] = {};
+        acc[resourceType].selected = false;
+        if (selectedResourceType === resourceType) {
+          acc[resourceType].selected = true;
+        }
+        Object.entries(resourceIdsGroupedByType[resourceType]).forEach(([subType, resourceIds]) => {
+          if (!acc[resourceType].subTypes) {
+            acc[resourceType].subTypes = {};
+          }
+          if (!acc[resourceType].subTypes[subType]) {
+            acc[resourceType].subTypes[subType] = {};
+          }
+          const subTypeResourceIds = Array.from(resourceIds);
+          acc[resourceType].subTypes[subType].resourceIds = subTypeResourceIds;
+          acc[resourceType].subTypes[subType].count = subTypeResourceIds.length;
+          const dateFilteredResourceIds = subTypeResourceIds
+            .filter((subTypeResourceId) => isResourceInDateRange(
+              resources[subTypeResourceId], activeDateStart, activeDateEnd,
+            ));
```

Comment:
there is already a selector for filtering all resources by date.  (and another that plucks relevant values for getting id and type)  this could be leveraged, but it would have to be the primary data Array this whole selector operates on, w/ grouping occurring, here

Reply:
This will have to be a refactor after the demo

Reply:
yes; I think there should be dedicated pass to trim / simplify the number of proliferating selectors in this file.